in-progress/7588-spartan-clusters.md

```diff
@@ -0,0 +1,301 @@
+apiVersion: chaos-mesh.org/v1alpha1
+kind: PodChaos
+metadata:
+  name: pod-failure-example
+  namespace: testing-network-namespace
+spec:
+  action: pod-failure
+  mode: one
+  duration: '30s'
+  selector:
+    labelSelectors:
+      'app.kubernetes.io/component': 'boot-node'
+```
+
+In reality those selectors and durations will be configurable via the `values.yaml` file.
+
+### +network tests
+
+The network tests will use an analogous helper to the `E2E_TEST` in `yarn-projects/end-to-end/Earthfile`.
+
+```dockerfile
+NETWORK_TEST:
+  FUNCTION
+  ARG hardware_concurrency=""
+  ARG namespace
+  ARG test
+  ARG network_values
+  ARG chaos_values
+  LOCALLY
+  # Let docker compose know about the pushed tags above
+  ENV AZTEC_DOCKER_TAG=$(git rev-parse HEAD)
+  # load the docker image into kind
+  RUN kind load docker-image aztecprotocol/end-to-end:$AZTEC_DOCKER_TAG
+  RUN helm install aztec-network helm-charts/aztec-network --set $network_values --namespace $namespace
+  RUN helm install aztec-chaos helm-charts/aztec-chaos --set $chaos_values --namespace $namespace
+  RUN helm test aztec-network --namespace $namespace
+```
+
+
+## Change Set
+
+Fill in bullets for each area that will be affected by this change.
+
+- [ ] Cryptography
+- [ ] Noir
+- [ ] Aztec.js
+- [x] PXE
+- [ ] Aztec.nr
+- [ ] Enshrined L2 Contracts
+- [ ] Private Kernel Circuits
+- [x] Sequencer
+- [ ] AVM
+- [ ] Public Kernel Circuits
+- [ ] Rollup Circuits
+- [ ] L1 Contracts
+- [ ] Prover
+- [ ] Economics
+- [ ] P2P Network
+- [x] DevOps
+
+## Test Plan
+
+### Token Transfer Matrix
+
+We will build out a matrix of conditions to test token transfers. This will include:
+
+- different network topologies (number of nodes, validators, pxes)
+- different network conditions (latency, node failures)
+
+Specifically, we will ensure that:
+
+- A block is proposed and added to the Pending Chain
+- A block is proven and added to the Proven Chain
+- A block is finalized
+- The network can tolerate a sequencer going offline
+- The network can tolerate a prover going offline
+- The network can tolerate a sequencer submitting an invalid block
+- The network can tolerate sequencers/validators with slow connections
+
+### Attack Scenarios
+
+We will verify:
+- A block proposed without a signature from the current proposer should fail
+- A prover submitting a proof with an invalid proof
+- A coalition of dishonest sequencers submitting a block with an invalid state transition
+- We can tolerate "soft" L1 censorship (i.e. the anvil instance is down for a period of time)
+
+### Existing e2e tests
+
+Existing e2e tests will continue to work as they do now. 
+
+We will gradually port the tests to work with either the existing setup or the new setup, configurable via an environment variable; this likely can be done by simply pointing the wallets that run in the tests to PXEs in the k8s cluster.
+
+## Documentation Plan
+
+We will write documentation on how people can join the `spartan` network. 
```

Comment:
I'm not familiar with k8s but all of this seems much easier than anything we have at the moment.